src/Memoize.jl

```diff
@@ -76,4 +78,9 @@ macro memoize(args...)
     end)
 
 end
+
+function memoize_cache(f::Function)
```

Comment:
```suggestion
function memoize_cache(f::Function)
    # This will fail in certain circumstances (eg. @memoize Base.sin(::MyNumberType) = ...) but I don't think there's 
    # a clean answer here, because we can already have multiple caches for certain functions, if the methods are 
    # defined in different modules.
```